src/boot.lisp

```diff
@@ -409,66 +411,126 @@
     `(multiple-value-call (lambda ,gvars ,@setqs)
        ,@form)))
 
+;;; types utils
+(defun object-type-code (object) (oget object "dt_Name"))
+(defun set-object-type-code (object tag) (oset tag object "dt_Name"))
```

Comment:
Also, this tagging system work for every value (including conses).  In other places of the source code, I have used a concept called "storage-vector`.  That is like the underlying structure to implement arrays in. and they support tags.

We could have something similar, l ike `storage-map` as well.  I think that might be more intuitive than this tagging scheme.